core/spec/models/spree/order_spec.rb

```diff
@@ -665,59 +665,136 @@ def compute(computable)
     end
   end
 
-  context "#associate_user!" do
-    let!(:user) { FactoryGirl.create(:user) }
-
-    it "should associate a user with a persisted order" do
-      order = FactoryGirl.create(:order_with_line_items, created_by: nil)
-      order.user = nil
-      order.email = nil
-      order.associate_user!(user)
-      expect(order.user).to eq(user)
-      expect(order.email).to eq(user.email)
-      expect(order.created_by).to eq(user)
-
-      # verify that the changes we made were persisted
-      order.reload
-      expect(order.user).to eq(user)
-      expect(order.email).to eq(user.email)
-      expect(order.created_by).to eq(user)
-    end
-
-    it "should not overwrite the created_by if it already is set" do
-      creator = create(:user)
-      order = FactoryGirl.create(:order_with_line_items, created_by: creator)
-
-      order.user = nil
-      order.email = nil
-      order.associate_user!(user)
-      expect(order.user).to eq(user)
-      expect(order.email).to eq(user.email)
-      expect(order.created_by).to eq(creator)
-
-      # verify that the changes we made were persisted
-      order.reload
-      expect(order.user).to eq(user)
-      expect(order.email).to eq(user.email)
-      expect(order.created_by).to eq(creator)
-    end
-
-    it "should associate a user with a non-persisted order" do
-      order = Spree::Order.new
+  describe "#associate_user!" do
+    let(:user) { FactoryGirl.create(:user_with_addreses) }
+    let(:email) { user.email }
+    let(:created_by) { user }
+    let(:bill_address) { user.bill_address }
+    let(:ship_address) { user.ship_address }
+    let(:override_email) { true }
+
+    let(:order) { FactoryGirl.build(:order, order_attributes) }
+
+    let(:order_attributes) do
+      {
+        user:         nil,
+        email:        nil,
+        created_by:   nil,
+        bill_address: nil,
+        ship_address: nil
+      }
+    end
 
-      expect do
-        order.associate_user!(user)
-      end.to change { [order.user, order.email] }.from([nil, nil]).to([user, user.email])
+    def assert_expected_order_state
```

Comment:
Assignment Branch Condition size for assert_expected_order_state is too high. [58/15]